src/common/com/intellij/plugins/haxe/model/HaxeModifiersModel.java

```diff
@@ -31,43 +32,44 @@ public HaxeModifiersModel(PsiElement baseElement) {
     this.baseElement = baseElement;
   }
 
-  public boolean hasModifier(HaxeModifierType modifier) {
+  public boolean hasModifier(@ModifierConstant String modifier) {
```

Comment:
Because modifier constant - is the type of modifier and basically it's a string, but HaxePsiModifier is a psi element, it can't be used in the methods `getModifier`, `hasModifier` and so on. Maybe I haven't got the question?

Reply:
Right. (Duh!)  The constants in HaxePsiModifier are Strings, too.  Maybe we should wrap those in an enum or sub-class of String.  (And, I guess we could do that later.)

As it is, I don't see how the @ModifierConstant marking enforces anything, given: a) @interface ModifierConstant has no code; and, b) it has not been applied to the constant strings in HaxePsiModifier.

Reply:
`@ModifierConstant` is annotation that enforcing to use specific constants, and prevent to passing wrong modifier, by checking values introduced by `@MagicConstant(stringValues)` (check `HaxePsiModifier`)

It's better to extract them into separate class and rename class to `HaxeModifierPsi` aswell.
But it's better to make this later.